src/hooks/useMaciSignup.ts

```diff
@@ -34,7 +45,16 @@ export const useMaciSignup = (): IUseMaciSignUpData => {
   }, [attestations]);
 
   const onSignup = useCallback(async () => {
-    if (!data?.publicKey || !signer || !attestationId) {
+    if (!data?.publicKey) {
+      setError(MaciSignUpError.NO_KEY);
+      return;
+    }
+    if (!signer) {
+      setError(MaciSignUpError.NO_SIGNER);
+      return;
+    }
+    if (!attestationId) {
+      setError(MaciSignUpError.NO_ATTESTATION);
       return;
     }
```

Comment:
These cases are not possible to get because you can't call signup if there is no attestation id, signer or public key.

Reply:
so it will look like this:

https://github.com/privacy-scaling-explorations/maci-rpgf/assets/9023842/3bad6765-de07-4bc8-ada0-7acb80172ec5

Reply:
If you don't have attestation signup button won't be shown, the same with signer if you don't connect the wallet, and with public key because it's created for the session.

Reply:
Showing modal for errors is also a bad ux pattern.